src/app/qgisapp.cpp

```diff
@@ -11319,6 +11324,10 @@ void QgisApp::activateDeactivateLayerRelatedActions( QgsMapLayer *layer )
       }
 
       mActionOpenFieldCalc->setEnabled( true );
+      mActionAddFeature->setText( addFeatureText );
+      mActionAddFeature->setToolTip( addFeatureText );
+      QgsGui::shortcutsManager()->unregisterAction( mActionAddFeature );
+      QgsGui::shortcutsManager()->registerAction( mActionAddFeature, mActionAddFeature->shortcut() );
```

Comment:
What's up with these actions?

Reply:
`registerAction` call updates action's tooltip.

I'd like to call [`QgsShortcutsManager::updateActionToolTip()`](https://github.com/qgis/QGIS/blob/f8f7d60cf0ac01bc36ccf1a5d038cce449034aad/src/gui/qgsshortcutsmanager.cpp#L305) method directly from qgisapp.cpp, so there wouldn't be need for unregistering/registering again, but such method is private. Do you think we can make it public just for such call?

Reply:
I think `updateActionToolTip()` should be a private slot, connected to each registered `QAction`'s `changed()` signal. Qt normally already implements recursivity checks in their setters. In case that's not there, we'll need to do that.

Reply:
That makes sense unless `changed()` is emitted too many times. I see [`changed()`](http://doc.qt.io/qt-5/qaction.html#changed) signal is also emitted when properties like enabled change. The enabled property of a number of actions changes each time a new layer type is selected in the layer tree.

If you don't think this would introduce too many unnecessary calls to `updateActionToolTip()`, I would be in favor of your solution.

Reply:
Yeah, too bad there's no dedicated signal.
I think this shouldn't be too much overhead if the method is cheap to execute in case nothing changed. The only potentially expensive operation is the regex stuff, to avoid that, there could be a dynamic `currentTooltip` property on the action introduced which is set at the end of the method and compared at the top of the method for an early exit.

Reply:
I gave this a second thought. I prefer the way it is done in the PR because it updates a single action and avoids to call unnecessarily the `updateActionToolTip()` method.

I actually attempted to implement what you suggested but found out that making an early comparison is a bit of a nightmare because the ToolTip gets some formatting (bold text) and an optional suffix (the shortcut). You can choose to store a base ToolTip with no format nor suffix (if you store the whole toolTip you'd need to get rid of formatting and the suffix, e.g., via Regular Expressions, which we want to avoid in this early stage), format it, append the suffix, and compare it against `action->toolTip()`. If they differ, you can use the string built on the base ToolTip and set it as the new ToolTip (you would need to block signals to do so). So far so good, but toolTip can be changed directly using `action->setToolTip("New")`, which will make the base ToolTip useless.

It would be easier if we can use, for instance, `action->text()` as the base string to build toolTips, but there are toolTips in the main GUI that don't correspond directly to its action's text (see, for instance, `mActionLocalCumulativeCutStretch`)

Could you give your suggestion a try? Otherwise, could you reconsider the PR? It's really much simpler.